src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.Integer.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            return TryFormatCore(value, 0xffffffff, buffer, out bytesWritten, format, encoder);
+        }
+
+        public static bool TryFormat(this ulong value, Span<byte> buffer, out int bytesWritten, TextFormat format = default(TextFormat), TextEncoder encoder = null)
+        {
+            return TryFormatCore(value, buffer, out bytesWritten, format, encoder);
+        }
+
+        public static bool TryFormat(this long value, Span<byte> buffer, out int bytesWritten, TextFormat format = default(TextFormat), TextEncoder encoder = null)
+        {
+            return TryFormatCore(value, 0xffffffffffffffff, buffer, out bytesWritten, format, encoder);
+        }
+
+        static bool TryFormatCore(long value, ulong mask, Span<byte> buffer, out int bytesWritten, TextFormat format, TextEncoder encoder)
+        {
+            encoder = encoder == null ? TextEncoder.Utf8 : encoder;
+            if (format.IsDefault || format.Symbol == 'g')
+            {
+                format.Symbol = 'G';
+            }
+
+            if (encoder.IsInvariantUtf8)
+                return TryFormatInvariantUtf8(value, mask, buffer, out bytesWritten, format);
+            else if (encoder.IsInvariantUtf16)
+                return TryFormatInvariantUtf16(value, mask, buffer, out bytesWritten, format);
+            else
+                return IntegerFormatter.TryFormatInt64(value, mask, buffer, out bytesWritten, format, encoder);
+        }
+
+        static bool TryFormatCore(ulong value, Span<byte> buffer, out int bytesWritten, TextFormat format, TextEncoder encoder)
+        {
+            encoder = encoder == null ? TextEncoder.Utf8 : encoder;
+
+            if (encoder.IsInvariantUtf8)
+                return TryFormatInvariantUtf8(value, buffer, out bytesWritten, format);
+            else if (encoder.IsInvariantUtf16)
+                return TryFormatInvariantUtf16(value, buffer, out bytesWritten, format);
+            else
+                return IntegerFormatter.TryFormatUInt64(value, buffer, out bytesWritten, format, encoder);
+        }
+
+        static bool TryFormatInvariantUtf8(long value, ulong mask, Span<byte> buffer, out int bytesWritten, TextFormat format)
+        {
+            switch (format.Symbol)
+            {
+                case (char)0:
+                case 'd':
+                case 'D':
+                case 'G':
+                case 'g':
+                    return InvariantUtf8IntegerFormatter.TryFormatDecimalInt64(value, format.Precision, buffer, out bytesWritten);
+
+                case 'n':
+                case 'N':
+                    return InvariantUtf8IntegerFormatter.TryFormatNumericInt64(value, format.Precision, buffer, out bytesWritten);
+
+                case 'x':
+                case 'X':
+                    return InvariantUtf8IntegerFormatter.TryFormatHexUInt64((ulong)value & mask, format.Precision, format.Symbol == 'x', buffer, out bytesWritten);
+
```

Comment:
Why not have a separate call for "x" and "X" cases, so you don't have to do the check twice (once for case match and once for format.Symbol=='x')